src/lib_db.ml

```diff
@@ -21,6 +21,10 @@ type t =
   }
 
 let rec internal_name_scope t ~dir =
+  if not (Path.is_descendant dir ~of_:t.anonymous_root) then (
```

Comment:
When this doesn't hold, we know that either the path is external either we'll end up reaching `Path.root`. So I suggest that we put this in the `None` case instead: `if Path.is_root path || not Path.is_local path then ...`. This will avoid the cost of `Path.is_descendant` for every lookup.

Reply:
OK, sure, but this does make the error worse as now we won't have the original path argument in the error message.